test/query.js

```diff
@@ -285,7 +285,7 @@ test('relation + super-relation on out-of-bbox node of a way', function (t) {
   })
 })
 
-test('opts.type: results sorted by type', function (t) {
+test.skip('opts.type: results sorted by type', function (t) {
```

Comment:
@gmaclennan I couldn't find anything in the [OSM "spec"](https://wiki.openstreetmap.org/wiki/API_v0.6#Retrieving_map_data_by_bounding_box:_GET_.2Fapi.2F0.6.2Fmap) on elements needing to come out in a certain order. Do you know why osm-p2p-server wants type-centric ordering?

Reply:
I skipped this test because I also didn't find this behavior in the spec, and to make it work I'd need to buffer all the results in memory. If we can avoid doing that, it would be preferable.